### DISCUSSION TITLE
libcrypto.so.1.1 not available in or for Linux Mint 21 (based on Ubuntu Linux 22.04.

### DISCUSSION THREAD

**Discussion by mattsmi** [pawelsalawa/sqlitestudio#4482] [Q&A]
### Details
Load SQLitestudio 3.3.3 and it comes up with two errors: it cannot load DbSqliteCipher, nor can it check for updates.

### Steps to reproduce
Run on Linux Mint 21 or Ubuntu Linux 22.04.

### Operating system
Linux Mint 21 or Ubuntu Linux 22.04.

### SQLiteStudio version
3.3.3

I cannot find the library anywhere in the repositories. There used to be a package libsll1.1, which I tried to install. It appears that the package has been made obsolete. The Calibre ebook library (calibre-ebook.com/) appears to install libcrypto.so.1.1 but holds it in a local library. I have copied it into the lib directory of SQLitestudio, but it does not seem to call it. Is there any way to make this work?
Many thanks in advance, Pawel!

**Comment by pawelsalawa** [comment]
If you copied the libcrypto.so.1.1 to the `lib/` subdir in SQLiteStudio, it generally should work. Can you please show me result of these commands executed from SQLiteStudio directory:
```
ls -l lib/*crypto*
ldd plugins/libDbSqliteCipher.so | grep crypto
```

**Comment by mattsmi** [comment]
Attached the screenshots.
When I manually add in libcrypto.so.1.1, one error disappears (it can now load DbSqliteCipher) and one remains (the one saying that it cannot update).

Cheers, Pawel!

![grepping](https://user-images.githubusercontent.com/6654733/183054505-6e5cd8b4-567a-4ef6-916c-92066bba6cff.png)
![error shown in SQLiteStudio](https://user-images.githubusercontent.com/6654733/183054508-d20cfb39-2f70-48a8-bcb6-f53aee61c882.png)

**Comment by pawelsalawa** [comment]
Okay, so here is some gap - you said:
> I have copied it into the lib directory of SQLitestudio, but it does not seem to call it.

but then I see:
![image](https://user-images.githubusercontent.com/481846/183056162-473e4cbb-cb92-4c4b-90da-e76a1f98d790.png)

I'm pointing this, because it should work if the file was there.

**Comment by mattsmi** [comment]
Sorry. A restore removed it. 
I added it back in, and as I said, only the first error was resolved, not the updating error.
Sorry for the confusion, Pawel.

**Reply by pawelsalawa** [reply]
Okay. Please open a web browser and go to https://sqlitestudio.pl. Then check whether your browser resolved a SSL certificate of the page correctly (a locker icon next to the page address).

**Comment by mattsmi** [comment]
Screenshot attached. SSL working fine in accessing web sites.

![SSL on Pawel web site](https://user-images.githubusercontent.com/6654733/183225198-6801f3e6-dbf2-4c26-94e6-7cada13bcff3.png)
![after libcrypto in lib](https://user-images.githubusercontent.com/6654733/183225201-a5305004-9de5-46ca-af14-a05fb6ad2974.png)

**Reply by pawelsalawa** [reply]
I'm out of ideas in that case. Sorry.
